### ISSUE TITLE
ProducesResponseType with typeof(IList<T>) in Response Messages doesn't serialize correctly

### ISSUE THREAD

**Issue by steamrolla** [domaindrivendev/Swashbuckle.AspNetCore#231]
Given a simple model, the `ProducesResponseType` attribute on an alternate response isn't serializing correctly.

```c#
[ProducesResponseType(typeof(IList<SimpleModel>), 200)]
```
Works excellently, no issues.

```c#
[ProducesResponseType(typeof(IList<SimpleModel>), 406)]
```
Results in `Array[object]` displaying in the `Response Messages` section.

**Comment by steamrolla** [comment]
Actually, I don't think error has anything to do with it.  It seems that any response listed in the `Response Messages` section has a `Response Model` of `Array[object]`.

**Comment by domaindrivendev** [comment]
@steamrolla - can you share what SimpleModel looks like?

**Comment by steamrolla** [comment]
Yup, here's the `ProducesResponseType` attribute from my source:

```c#
[ProducesResponseType(typeof(IList<FieldValidationError>), 406)]
```

And `FieldValidationError`:
```c#
public class FieldValidationError
{
    public string Field { get; set; }
    public string Error { get; set; }
}
```

**Comment by domaindrivendev** [comment]
Swashbuckle generates Swagger JSON from your routes, controllers and models and then, for convenience, provides an embedded version of the [swagger-ui](https://github.com/swagger-api/swagger-ui) project. In this case, the generated Swagger JSON looks 100% correct so this is fundamentally an issue with the way in which the swagger-ui renders complex types in the Responses section.

So, in conclusion, this would have to be fixed by the swagger-ui folks.

**Comment by adamkjonesmesh** [comment]
Does anyone know if there has been a resolution on this issue?  I am experiencing the exact same problem.  Given a ProducesResponseType(typeof(List<SomeComplexObject>), 500) the resulting UI generates Array[object].

**Comment by kierenj** [comment]
Aha - think I'm having this issue too.  Very similar use-case to the OP in fact.

I found this from 2015 - https://github.com/domaindrivendev/Swashbuckle/issues/200 - same or very similar issue?  Should I/we raise a ticket with swagger-ui?

**Comment by kierenj** [comment]
Raised an issue here - https://github.com/swagger-api/swagger-ui/issues/4079

**Comment by kierenj** [comment]
From https://github.com/swagger-api/swagger-ui/issues/4079#issuecomment-356899349 it looks like the latest Swagger UI version can handle this.

How do I go about updating the version please?

**Comment by kierenj** [comment]
Ok, added the feed, using SwaggerUI3, all working fine for me.

**Comment by domaindrivendev** [comment]
Fixed as of release 2.0.0, which includes swagger-ui 3.x.
